### PR TITLE
py-ipyparallel: add 7.1.0

### PR DESCRIPTION
**Pull request by genric**
Also take releases from pypi.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-ipyparallel/package.py [hunk 1]
```diff
@@ -10,20 +10,30 @@ class PyIpyparallel(PythonPackage):
...
-    depends_on('py-decorator', type=('build', 'run'))
-    depends_on('py-pyzmq@13:', type=('build', 'run'))
-    depends_on('py-traitlets@4.3:', type=('build', 'run'))
-    depends_on('py-ipython@4:', type=('build', 'run'))
-    depends_on('py-jupyter-client', type=('build', 'run'))
-    depends_on('py-ipykernel@4.4:', type=('build', 'run'))
+    depends_on('python@3.6:', type=('build', 'run'), when='@7.1.0:')
+
+    depends_on('py-setuptools',        type='build')
+    depends_on('py-jupyter-packaging', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ipyparallel/package.py:24]: I don't think this dependency is actually needed. The only reference I see is that `setup.py` imports `setupbase.py` which was generated by `jupyter-packaging`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-ipyparallel/package.py (hunk 1 above)
Comment (genric) [reply var/spack/repos/builtin/packages/py-ipyparallel/package.py:24]: It was failing without it when doing `spack install ...` in setupbase.py:
```
    from packaging.version import VERSION_PATTERN
ModuleNotFoundError: No module named 'packaging'
```

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-ipyparallel/package.py:24]: Ah, then just add a build dep on `py-packaging`? Not sure why they didn't add that to `setup_requires` in `setup.py` but it's in the `pyproject.toml`.

Comment (genric) [reply var/spack/repos/builtin/packages/py-ipyparallel/package.py:24]: With `py-packaging` I get this:
```
setupbase.py, line 128, in _make_wrapper
    class _Wrapped(klass):
TypeError: NoneType takes no arguments
```

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-ipyparallel/package.py:24]: Hmm, I don't see what could cause that. `py-jupyter-packaging` isn't aliasing `py-packaging` or anything. Let's keep it your way for now.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**